src/test/java/com/synopsys/integration/stepworkflow/StepWorkflowTest.java

```diff
@@ -305,30 +228,31 @@ public void testConditionalButOnlyIfFalse() {
 
         // Call run() and validate return StepWorkflowResponse
         StepWorkflowResponse<Integer> response = buildStepWorkflow.run();
-        assertTrue(response.wasSuccessful());
-        assertEquals(TEST_DATA_THREE, response.getData());
-        assertNull(response.getException());
+        assertFalse(response.wasSuccessful());
+        assertNull(response.getData());
+        assertEquals(integrationException, response.getException());
     }
 
-    @Test
-    public void testConditionalFailedPreviousResponse() {
-        // Create and validate first() Builder using unsuccessfulDataSupplier
-        assertEquals(unsuccessfulSupplierOne, unsuccessfulBuilderOfSupplierOne.start.step);
-        assertNull(unsuccessfulBuilderOfSupplierOne.start.next);
-        assertEquals(unsuccessfulSupplierOne, unsuccessfulBuilderOfSupplierOne.end.step);
-
+    @ParameterizedTest
+    @MethodSource("runConditionalWorkflowObjectToTest")
```

Comment:
Since we're not testing the null case here (which is very reasonable), @ ValueSource(booleans = { true, false }) is probably clearer than @ MethodSource